functests/xt_u32/xt_u32.go

```diff
@@ -0,0 +1,223 @@
+package xt_u32
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	igntypes "github.com/coreos/ignition/config/v2_2/types"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/client"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/discovery"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/execute"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/images"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/namespaces"
+	utilNodes "github.com/openshift-kni/cnf-features-deploy/functests/utils/nodes"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/pods"
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const hostnameLabel = "kubernetes.io/hostname"
+const TestNamespace = "test-xt-u32"
+
+const xt_u32LoadPath = "/etc/modules-load.d/xt_u32-load.conf"
+
+var (
+	xt_u32NodeSelector string
+	hasNonCnfWorkers   bool
+)
+
+func init() {
+	roleWorkerCNF := os.Getenv("ROLE_WORKER_CNF")
+	if roleWorkerCNF != "" {
+		xt_u32NodeSelector = fmt.Sprintf("node-role.kubernetes.io/%s=", roleWorkerCNF)
+	}
+
+	hasNonCnfWorkers = true
```

Comment:
This was for the negative test, but I don't see it here.
I think this is not needed.

Reply:
but adding one can be a good defensive approach
otherwise we might miss changes like the module loading by default (due to some RHEL/RHCOS change or if OKD/OCP is ever supported on other oses)
it is less important, but still something we'd like to check for.

so I guess it's ok to remove from this commit, but add on a separate commit that will add the negative test.

Reply:
Agree, also the SCTP approach of having the value injected from outside revealed to be a bit cluncky (especially when somebody else needs to run these tests), so in that case automatic discovery of non SCTP / XT_U32 workers would be better